src/Symfony/Component/Routing/Tests/Fixtures/dumper/url_matcher12.php

```diff
@@ -0,0 +1,91 @@
+<?php
+
+use Symfony\Component\Routing\Exception\MethodNotAllowedException;
+use Symfony\Component\Routing\Exception\ResourceNotFoundException;
+use Symfony\Component\Routing\RequestContext;
+
+/**
+ * This class has been auto-generated
+ * by the Symfony Routing Component.
+ */
+class ProjectUrlMatcher extends Symfony\Component\Routing\Matcher\UrlMatcher
+{
+    public function __construct(RequestContext $context)
+    {
+        $this->context = $context;
+    }
+
+    public function match($rawPathinfo)
+    {
+        $allow = array();
+        $pathinfo = rawurldecode($rawPathinfo);
+        $trimmedPathinfo = rtrim($pathinfo, '/');
+        $context = $this->context;
+        $requestMethod = $canonicalMethod = $context->getMethod();
+
+        if ('HEAD' === $requestMethod) {
+            $canonicalMethod = 'GET';
+        }
+
+        $matchedPathinfo = $pathinfo;
+        $regexList = array(
+            0 => '{^(?'
+                    .'|/abc([^/]++)(?'
+                        .'|/1(?'
+                            .'|(*:27)'
+                            .'|0(?'
+                                .'|(*:38)'
+                                .'|0(*:46)'
+                            .')'
+                        .')'
+                        .'|/2(?'
+                            .'|(*:60)'
+                            .'|0(?'
+                                .'|(*:71)'
+                                .'|0(*:79)'
+                            .')'
+                        .')'
+                    .')'
```

Comment:
without this change, a list is generated instead of a tree:
```
                        .'|/2(*:33)'
                        .'|/10(*:43)'
                        .'|/20(*:53)'
                        .'|/100(*:64)'
                        .'|/200(*:75)'
```